kubeflow/profiles/tests/profiles_test.py

```diff
@@ -0,0 +1,81 @@
+"""Test Profiles custom resource.
+
+todo
+
+"""
+
+import logging
+import os
+import subprocess
+import re
+import requests
+from retrying import retry
+import six
+import time
+
+import pytest
+
+from kubernetes.config import kube_config
+from kubernetes import client as k8s_client
+from kubeflow.testing import util
+
+GROUP = "kubeflow.org"
+PLURAL = "profiles"
+KIND = "Profile"
+VERSION = "v1alpha1"
+
+logging.basicConfig(level=logging.INFO,
+                    format=('%(levelname)s|%(asctime)s'
+                            '|%(pathname)s|%(lineno)d| %(message)s'),
+                    datefmt='%Y-%m-%dT%H:%M:%S',
+                    )
+logging.getLogger().setLevel(logging.INFO)
+
+def test_profiles():
+  app_credentials = os.getenv("GOOGLE_APPLICATION_CREDENTIALS")
+  if app_credentials:
+    logging.info("Activate service account")
+    util.run(["gcloud", "auth", "activate-service-account",
+              "--key-file=" + app_credentials])
+
+  # util.load_kube_config appears to hang on python3
+  kube_config.load_kube_config()
+  api_client = k8s_client.ApiClient()
+
+  this_dir = os.path.dirname(__file__)
+  util.run(["kubectl", "apply", "-f", "sample_profile.yaml"], cwd=this_dir)
+
+  # TODO: check CR status/condition instead of sleep
+  # conditions = ["Ready"]
+  # namespace = "kubeflow"
+  # name = "john"
+  # results = util.wait_for_cr_condition(api_client, GROUP, PLURAL, VERSION,
+  #                                      namespace, name, conditions)
+  # logging.info("Result of CRD:\n%s", results)
+  time.sleep(5)
+
+  # Verifies the namespace is created.
+  name = "john"  # The name of the profile, also the new namespace's name.
+  coreV1 = k8s_client.CoreV1Api(api_client)
+  resp = coreV1.read_namespace(name)
+  logging.info("found namespace: %s", resp)
+
+  rbacV1 = k8s_client.RbacAuthorizationV1Api(api_client)
+  resp = rbacV1.read_namespaced_role("edit", name)
+  logging.info("role: %s", resp)
+  resp = rbacV1.read_namespaced_role_binding("default", name)
+  logging.info("role binding: %s", resp)
+
+  # delete the profile and make sure namespace is deleted
+  util.run(["kubectl", "delete", "-f", "sample_profile.yaml"], cwd=this_dir)
+  resp = coreV1.read_namespace(name)
+  logging.info("read namespace response: %s", resp)
```

Comment:
Shall we verify the namespace is deleted after profile CR is gone?
Also might need to add retry/sleep in case namespace is not deleted immediately.

Reply:
Yes, thank you @kunmingg .
I was still testing what's the behavior of read_namespace if it's not found.
It's not clear to me from the doc: https://github.com/kubernetes-client/python/blob/master/kubernetes/docs/CoreV1Api.md#read_namespace

Will let you know when it's ready